builder/azure/arm/config.go

```diff
@@ -106,6 +106,8 @@ type SharedImageGalleryDestination struct {
 	// Specify a storage account type for the Shared Image Gallery Image Version.
 	// Defaults to `Standard_LRS`. Accepted values are `Standard_LRS`, `Standard_ZRS` and `Premium_LRS`
 	SigDestinationStorageAccountType string `mapstructure:"storage_account_type"`
+	// Set to true if publishing to a Specialized Gallery, this skips a call to set the resulting VM's OS state as Generalized
+	SigDestinationSkipGeneralization bool `mapstructure:"skip_generalization"`
```

Comment:
What do you think about calling this attribute specialized to match the [Azure CLI flag](https://learn.microsoft.com/en-us/azure/virtual-machines/vm-specialized-image-version?tabs=cli%2Ccli2%2Ccli3%2Ccli4%2Ccli5#create-a-vm-from-your-gallery)? 

The behavior is the same as it is written the only change is the name of the configuration attribute. 
```suggestion
	SigDestinationSkipGeneralization bool `mapstructure:"specialized"`
```